### PR TITLE
Add getPut for IncomingRequest 

### PR DESCRIPTION
**Pull request by wuzheng40**
Add function to get Data from PUT method.
ex:
$request = \Config\Services::request();
$data = $request->getPut();

### CODE REVIEW

**Comment by jim-parry** [comment]
A couple of points, sorry:
1) Your method description, implying getting a single item, is at odds with the implementation, returning all PUT parameters
2) Why deviate from the other "get" method signatures? Shouldn't you provide an index and filter parameter too?
3) Would this be better or also addressed in Request:fetchGlobal?
4) unit testing for this proposed change?
5) user guide update for this proposed change?
6) GPG-signed commits for code?
7) This is more than a single file change, and you might find a feature branch (at your end) easier to manage this with.
@lonnieezell What do you think?

**Comment by lonnieezell** [comment]
This one's a tricky one. PUT data is quite different from GET or POST, and is more like the `getJSON()` method that already exists. I think the issue is the name of method, honestly, because there's nothing specific about the data that makes it come from a PUT request, as compared to a DELETE request, PATCH, etc. 

I think the most logical solution is to rename the method to something like `getInput`, though that name could cause confusion with the `getOldInput` method since it's retrieving GET/POST data. It does make sense, though, since we're basically pulling directly from `php::/input`. 

That method would be a good balance with the `getJSON` method, also, which simply provides a convenience method to get the raw input already. 

This would need tests and userguide updates to accompany it, though.

**Comment by wuzheng40** [comment]
I modify fetchGlobal at first, but when I find getJSON function in IncomingRequest, I decide to add new function in it too. It`s similar with getJSON，but has different return only @jim-parry

getRawParam ? @lonnieezell

**Comment by lonnieezell** [comment]
@wuzheng40 I'd go for `getRawInput()`.

**Comment by wuzheng40** [comment]
@lonnieezell Ok, I renamed the function to getRawInput.

**Comment by lonnieezell** [comment]
@wuzheng40 Thanks for that. 

Before I can accept this, I also need the following items: 

1) add relevant tests to [IncomingRequestTest](https://github.com/bcit-ci/CodeIgniter4/blob/develop/tests/system/HTTP/IncomingRequestTest.php)
2) Add information about this to [the docs](https://github.com/bcit-ci/CodeIgniter4/blob/develop/user_guide_src/source/libraries/incomingrequest.rst)
3) We ask that all contributors GPG sign their commits. It's relatively easy to setup. Here's [instructions](https://bcit-ci.github.io/CodeIgniter4/contributing/signing.html)

**Comment by wuzheng40** [comment]
@lonnieezell Tests and docs have been added. I add a GPG sign in my setting, Please check it , thanks.

File: user_guide_src/source/libraries/incomingrequest.rst [hunk 1]
```diff
@@ -133,6 +133,14 @@ arrays, pass in ``true`` as the first parameter.
 The second and third parameters match up to the ``depth`` and ``options`` arguments of the
 `json_decode <http://php.net/manual/en/function.json-decode.php>`_ PHP function.
 
+**Getting Raw data (PUT, PATCH, DELETE)**
+
+Finally, you can grab the contents of php://input as a raw stream with ``getRawInput()``.
```
Comment (lonnieezell) [inline user_guide_src/source/libraries/incomingrequest.rst]: You can't get a stream back from the method, so this is a little misleading.

File: user_guide_src/source/libraries/incomingrequest.rst [hunk 2]
```diff
@@ -133,6 +133,14 @@ arrays, pass in ``true`` as the first parameter.
...
+
+	$data = $request->getRawInput();
+
+By default, this will convert param string to an array.
```
Comment (lonnieezell) [inline user_guide_src/source/libraries/incomingrequest.rst]: "By default" implies it's possible to get it back in a different format, but currently we can only retrieve it as an array.

If you could show a small example of what they should expect to get, that would be awesome. I need to add that to the getJSON docs also, it looks like.

File: user_guide_src/source/libraries/incomingrequest.rst (hunk 1 above)
Comment (wuzheng40) [inline user_guide_src/source/libraries/incomingrequest.rst]: You meanings is Getting => Retrieving ?

**Comment by wuzheng40** [comment]
@lonnieezell I added example for getRawInput, and modify docs, please check, thanks

File: user_guide_src/source/libraries/incomingrequest.rst [hunk 3]
```diff
@@ -133,6 +133,24 @@ arrays, pass in ``true`` as the first parameter.
...
+
+Finally, you can grab the contents of php://input as a raw stream with ``getRawInput()``.
+
+	$data = $request->getRawInput();
+
+This will retrieve data and convert it to an array. Like this::
+	<form>
+		<input type="text" name="Username" value="Admin">
+		<input type="text" name="Role" value="Administrator">
+		<input type="hidden" name="_method" value="PUT">
```
Comment (lonnieezell) [inline user_guide_src/source/libraries/incomingrequest.rst]: We don't support method spoofing like this... please remove.

Comment (wuzheng40) [inline user_guide_src/source/libraries/incomingrequest.rst]: I removed these. Please check it.

**Pull request merged**